src/Shared/FileSystem/ManagedFileSystem.cs

```diff
@@ -38,19 +39,64 @@ public byte[] ReadFileAllBytes(string path)
             return File.ReadAllBytes(path);
         }
 
+#if FEATURE_MSIOREDIST
+        private IEnumerable<string> HandleFileLoadException(
+            Func<string, string, Microsoft.IO.SearchOption, IEnumerable<string>> enumerateFunctionDelegate,
+            string path,
+            string searchPattern,
+            Microsoft.IO.SearchOption searchOption
+        )
+        {
+            try
+            {
+                return enumerateFunctionDelegate(path, searchPattern, searchOption);
+            }
+            // Microsoft.IO.Redist has a dependency on System.Buffers and if System.Buffers assembly is not found the line above throws an exception.
+            // However, FileMatcher class (that in most cases calls the enumeration) does not allow to fail on a IO-related exception. Such behavior hides the actual exception and makes it obscure.
+            // We rethrow it to make it fail with a proper error message and call stack.
+            catch (FileLoadException ex)
+            {
+                throw new InvalidOperationException("Could not load file or assembly.", ex);
```

Comment:
So, when the exception goes outside MSBuild we always want it to be localized? I will update it here in this case. 
I did not localize it because I do not expect it to be seen by many. But this exception indeed goes outside MSBuild. It is actually the main purpose of this exception: not to be caught.

Reply:
> So, when the exception goes outside MSBuild we always want it to be localized?

Not sure how much we care. I see examples of both localized and non-localized exception messages in the source tree. Maybe bring it up in today's PR reviews call?